### DISCUSSION TITLE
MAC BETA TESTERS NEEDED!!!

### DISCUSSION THREAD

**Discussion by phnxdrgn536** [phnxdrgn536/FLSun-v400#17] [Announcements]
Anyone wanting to help test a terminal program for mac OSX to automatically download and install the profiles, please post your interest here so that we can get you started.

**Comment by MidMitten3D** [comment]
I'm ready to give this a try. My printer is mostly operational again, but I'm waiting on a new effector assembly from Flsun to arrive. Mine was very badly put together and damaged on arrival. I think that's where most of my problems have stemmed from.

**Reply by bgeorgeiii** [reply]
The testing we need help with doesn't even require a printer. We're only trying to confirm the Cura profiles load properly on MacOS.

**Reply by MidMitten3D** [reply]
ok, I'm still new to this side of things. I was able to get the original posted profiles loaded by dragging the files into the corresponding files in the cura config folder. Is this a similar process?

**Reply by phnxdrgn536** [reply]
sort of... it is a simpler process... once the loader is fully functional it will automatically download and place the files in the proper folder. this also means that by running the loader subsequent times it will automatically update the files you already have with any updates or changes made to the files in the GitHub.

**Reply by phnxdrgn536** [reply]
https://github.com/bgeorgeiii/FLSun-v400/raw/loader-updates/Loader/Releases/CuraProfileLoader-0.0.2.zip

the loader can be found here

**Reply by MidMitten3D** [reply]
Bear with me. what do I do with the file after download? I tried to import as a profile but that didn't work.

**Reply by phnxdrgn536** [reply]
extract the files to a folder and run the exe in the osx folder... it should create a log file in that folder that will give us feedback if we got the file structure correct and if that's the case we can move forward with it getting the actual files

**Reply by bgeorgeiii** [reply]
> Bear with me. what do I do with the file after download? I tried to import as a profile but that didn't work.

More specifically, unzip the file and run "CuraProfileLoader" under the osx-x64 folder.

**Reply by MidMitten3D** [reply]
Okay, I was confused because after running it just opened a text file full of gibberish. 
Do I need to do anything more?

**Reply by phnxdrgn536** [reply]
yes post the txt file here

**Comment by MidMitten3D** [comment]
[CuraProfileLoader.txt](https://github.com/phnxdrgn536/FLSun-v400/files/10438724/CuraProfileLoader.txt)
sorry it took soo long to figure out with the kids running around lol

**Reply by bgeorgeiii** [reply]
I'm not really sure what's happened / gone wrong here but this is definitely not the expected output. How exactly did you open / run the app? Also which version of macOS are you on?

If possible can you attempt to run it via Terminal. Open the terminal and navigate to the folder where you extracted CuraProfileLoader from the osx-x64 directory and run it that way?

Ideally the application shouldn't open anything, it should just log to a file called output-{date}.txt placed next to the application.

**Reply by MidMitten3D** [reply]
MacOS 13.1
I downloaded the file. opened the osx-x64 folder. double clicked the file. clicked Open and then the text file pops up on my screen.
![IMG_3298](https://user-images.githubusercontent.com/122140225/213042904-eda4a5d6-afd3-48ec-aa74-d691f1e5cc30.jpeg)
![IMG_3299](https://user-images.githubusercontent.com/122140225/213042922-46a5512a-6e0a-4dd1-8ea0-17fa7884fdbd.jpeg)
![IMG_3300](https://user-images.githubusercontent.com/122140225/213042934-7669a476-0fe2-4226-b688-d9114b0ba52b.jpeg)

I don't really know how to navigate in the Mac terminal to be honest
![IMG_3302](https://user-images.githubusercontent.com/122140225/213047396-a2fdf475-b01d-4338-9744-9eecc4716e6f.jpeg)

**Reply by MidMitten3D** [reply]
quick bit of Youtube university got me to where I think I should be in terminal. But I get the same result when opening. should I use a different command?
![IMG_3304](https://user-images.githubusercontent.com/122140225/213054494-02709b9b-7613-43bc-a717-d7bbf51ee4a7.jpeg)

**Comment by phnxdrgn536** [comment]
Executables located in directories that are not on the path cannot be run
unless you specify their full location. So in your case, assuming the
executable is in the current directory you are working with, then you can
execute it as such

./my-exec

Where my-exec is the name of your program.

**Reply by MidMitten3D** [reply]
sorry, I'm really lost now. I extracted the downloaded folder to my desktop for ease of finding in terminal. The file in the osx-x64 folder is not showing up as an exe or dmg though. I'm in that folder's directory in terminal but the only commands that have worked are "open curaprofileloader" and "open ./curaprofileloader" , which just opens the text file again.

**Comment by phnxdrgn536** [comment]
The same place you type “open curaprofileloader” type “./curaprofileloader”

**Reply by MidMitten3D** [reply]
When I do that I get
zsh: permission denied: ./curaprofileloader

**Reply by bgeorgeiii** [reply]
It sounds like we're going to have to do some more work to publish the macOS version than simply building / distributing it.

**Reply by MidMitten3D** [reply]
Hopefully you get someone else with more experience than me to test this. I've only been in this realm for the past 2 months while trying to figure out my v400. Spent my first year of printing just slicing in cura, and printing from a flash drive on my Elegoo Neptune x with everything stock. That thing is wearing out now and needs some upgrades but there's really nothing out there specifically for it. so I've been diving deep. 
Learning a lot and hope I'm more of a help than a setback lol.

**Comment by phnxdrgn536** [comment]
Try one more time by typing “sudo ./curaprofileloader” if that does not
work for you then you don’t have the proper user security to run as admin.

**Reply by MidMitten3D** [reply]
I double checked what I know to look for, I am the computer admin, have full permissions, and have nothing blocked. I also tried to turn off firewall and still got the same result.

dejai@Dejais-MacBook-Air osx-x64 % sudo ./curaprofileloader
Password:
sudo: ./curaprofileloader: command not found
dejai@Dejais-MacBook-Air osx-x64 % 

I then ran "sudo open ./curaprofileloader" but it opened the same txt file as the previous times

**Reply by bgeorgeiii** [reply]
I think I need to mark the file as executable. It's possible you can do the same thing from your end if you'd like to try. In the terminal at the directory where CuraProfileLoader is, you can attempt to run 'chmod +x CuraProfileLoader' without the quotes. Then just '.\CuraProfileLoader' in that same window. 

Regardless I'll also update the file on my side so other users don't have to do it going forward.

**Reply by MidMitten3D** [reply]
dejai@Dejais-MacBook-Air osx-x64 % chmod -x.\CuraProfileLoader
usage:	chmod [-fhv] [-R [-H | -L | -P]] [-a | +a | =a  [i][# [ n]]] mode|entry file ...
	chmod [-fhv] [-R [-H | -L | -P]] [-E | -C | -N | -i | -I] file ...
dejai@Dejais-MacBook-Air osx-x64 %

**Reply by MidMitten3D** [reply]
I seen where I messed up. I think I got it to work now. 

Last login: Wed Jan 18 09:09:21 on ttys000
/Users/dejai/Desktop/CuraProfileLoader/osx-x64/CuraProfileLoader ; exit;        
dejai@Dejais-MacBook-Air ~ % /Users/dejai/Desktop/CuraProfileLoader/osx-x64/CuraProfileLoader ; exit;
Select Firmware: (Use arrow keys to select, Enter to submit.)
    [x] stock
    [ ] Reflashed

Select Cura Version: (Use arrow keys to select, Enter to submit.)
    [ ] 5.0                                                                     
    [ ] 5.1                                                                     
    [ ] 5.2                                                                     
    [x] 5.3                                                                     
    [ ] 5.4                                                                     

All set.
Press any key to exit.

**Reply by bgeorgeiii** [reply]
AYYYYYYY. That looks like it!. You should notice an output-{date}.txt next to the app. Send us that log.

**Reply by MidMitten3D** [reply]
after running the proper command 'chmod +x CuraProfileLoader' and returning to the original folder on my desktop. The 'curaprofileloader' changed to an .exe icon. I clicked that and it ran the script.

**Reply by MidMitten3D** [reply]
It did not create anything in the directory after running. I tried to go back and do it from the terminal and it ran just the same but I don't see a new txt file in the directory

**Reply by MidMitten3D** [reply]
![IMG_3308](https://user-images.githubusercontent.com/122140225/213207520-085ebdef-081c-4611-9813-c283049cb920.jpeg)

**Reply by bgeorgeiii** [reply]
Yea I had the wrong chmod command listed originally. I ninja edited the post but wasn't sure if you had seen it or not. 

I'm not sure where it would have placed your output but I'll look into that and see if we can find it.

**Reply by MidMitten3D** [reply]
ah, I figured I just did it wrong haha.  but noticed the + after looking back on here. I've ran it a couple times now but can't find any new files. feels like we're making progress though.

**Comment by phnxdrgn536** [comment]
Hey type “history” in the terminal and see if it shows the output file or
it’s location or if it just shows commands history

**Reply by MidMitten3D** [reply]
nope, just shows me a list of previous commands

**Comment by phnxdrgn536** [comment]
👍 thought so

**Reply by MidMitten3D** [reply]
been doing searches in finder to find anything created this morning after running the script multiple times but can't find anything. 
I have some running around to do, I'll check back later to see if you guys have come up with a solution.

**Reply by MidMitten3D** [reply]
I'm getting the same weird txt file pop up when I try to run the new loader. not sure what I'm doing wrong or if it's some kind of admin setting that I don't have enabled.

**Reply by MidMitten3D** [reply]
using terminal I was able to run it by going to the directory and running 'chmod +x CuraProfileLoader' again and then
'open curaprofileloader' 
but now the v400.stl mesh is weird looking.

**Reply by MidMitten3D** [reply]
replaced the mesh stl with the one from the original batch files in the zip file and It's back to normal. I'm not sure if the loader worked but it's showing that modifications were made to some files in the cura config folder after running.

**Comment by phnxdrgn536** [comment]
The profiles I believe are downloading properly, however we have an issue
of the STL downloading with the wrong encoding.

**Reply by MidMitten3D** [reply]
not bad for how quickly you guys got this together! How will we know going forward if there was an update to a profile? aside from checking the cura config folder for modified dates.

**Comment by phnxdrgn536** [comment]
Keep an eye on the discussions I’ll post any major updates or changes…
otherwise they will just be minor adjustments to values in the profiles.

**Reply by MidMitten3D** [reply]
also, just FYI. I did not start with another fresh install of cura before doing this. I no longer had the original supplied profile installed after switching to your stuff. But I was able to retain my custom profiles as well as your new updates. using cura aplah xmas 5.3.0
just in case anyone was worried about loosing their customs profiles after running the loader.

**Comment by phnxdrgn536** [comment]
👍 yep the only issues we have is when the original FLSun files are still
in cura
